wercker/wercker
# docker-run-auth

User should enter below parameters in wercker.yml during internal/docker-run, If Image is from private repository.

    - internal/docker-run:
      image: $USER/docker-build-golang
      name: myTestContainer
      username: $DOCKER_USERNAME
      password: $DOCKER_PASSWORD
      registry: $DOCKER_RGISTRY

Stage Tests

01) Run image from private repository. Provide required credentials(Result=PASS)
https://dev.wercker.com/gauagnih/docker-run/runs/build/5b503f74adf509000662465a
=================================================================

02) Run image from private repository. Make sure box image is also private and from another repository. Provide required credentials. (Result=PASS)
https://dev.wercker.com/gauagnih/docker-run/runs/build/5b506959adf50900066247b3
=================================================================

03) Run public image.(Result=PASS)
https://dev.wercker.com/gauagnih/docker-run/runs/build/5b504355adf509000662467f 
=================================================================

04) Run public image with credentials specified of private repository.(Result=PASS)
https://dev.wercker.com/gauagnih/docker-run/runs/build/5b5043d7adf509000662468f
=================================================================

05) Run public image with wrong credentials.(Result=PASS) 
https://dev.wercker.com/gauagnih/docker-run/runs/build/5b505ad7adf5090006624717?step=5b505aec84b94f000777c6e3
=================================================================

FAIL scenarios 
06) Run image from private repository. Credentials not passed for private image.(Result=FAIL)
https://dev.wercker.com/gauagnih/docker-run/runs/build/5b504254adf5090006624669?step=5b50426884b94f000777c347
=================================================================

07) Run image from private repository. Registry not correct.(Result=FAIL)
https://dev.wercker.com/gauagnih/docker-run/runs/build/5b504254adf5090006624669?step=5b50426884b94f000777c347
=================================================================

08) Run public image and provide wrong registry.(Result=FAIL)
https://dev.wercker.com/gauagnih/docker-run/runs/build/5b5045aaadf50900066246b1?step=5b5045be3ff5480007e9c90e
